### PR TITLE
feat(Menu): Add Menu component

### PR DESCRIPTION
**Pull request by lkappeler**
References https://autoricardo.atlassian.net/browse/VSST-412
Design: https://www.figma.com/file/8vy6tgk0DV9KSKyJBjiLSH/12.-AS24---Web---Patterns

- [x] Get feedback from Designer for the min-width

## Motivation and context

Adds the Menu component

## Before

N/A

## After

Screenshot 2022-10-10 at 16 01 14

## How to test

http://localhost:6006/?path=/story/patterns-data-display-menu--overview&args=title:Hello

### CODE REVIEW

File: src/components/menu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  items: MenuItem[];
+}
+
+const Menu: FC<MenuProps> = ({ title, items }) => {
+  return (
+    <ChakraMenu>
+      <MenuButton
+        as={Button}
+        padding={0}
+        rightIcon={<ChevronIcon boxSize="xs" />}
```
Comment (talamcol) [inline src/components/menu/index.tsx:28]: I cannot figure it out a 100% with the design you have but I think the common pattern is to rotate this chevron when the menu is open?

File: src/components/menu/index.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs';
+
+import Menu from './index.tsx';
+
+<Meta title="Patterns/Data Display/Menu" component={Menu} />
+
+export const Template = ({ title }) => {
+  const dummyOnClickHandlerFactory = (value) => {
+    return () => {
+      alert(`${value} clicked`);
```
Comment (talamcol) [inline src/components/menu/index.stories.mdx:10]: could we use the storybook actions instead of an alert?

File: src/themes/components/menu.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  minW: '3xs',
+  py: '2',
+  zIndex: 1,
+  borderRadius: 'sm',
+  borderWidth: '1px',
+  borderColor: 'gray.200',
+});
+
+const baseStyleItem = defineStyle({
+  paddingY: 'md',
```
Comment (talamcol) [inline src/themes/components/menu.ts:23]: in the design there is a hover color for the user menu element that is blue. Was this dropped?
image

**Review by talamcol — APPROVED** [review]
two small design questions (but the Figma page is messy so I am not sure if I checked the right thing). Code wise it looks good!

File: src/themes/components/menu.ts (hunk 3 above)
Comment (lkappeler) [reply src/themes/components/menu.ts:23]: It's intended for the language navigation I can have a chat with Sergiu regarding the rotating chevron. For the user navigation, we will make use of the drawer component (and the design will slightly change)
PoC: https://github.com/smg-automotive/components-pkg/pull/187
Designs: https://www.figma.com/file/8vy6tgk0DV9KSKyJBjiLSH/12.-AS24---Web---Patterns?node-id=159%3A5729

File: src/components/menu/index.tsx (hunk 1 above)
Comment (lkappeler) [reply src/components/menu/index.tsx:28]: see: https://github.com/smg-automotive/components-pkg/pull/277#discussion_r991916015

**Review by boyarskiy — APPROVED** [review]
Looks good

**Pull request merged**